theano/tensor/signal/pool.py

```diff
@@ -629,14 +637,15 @@ def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
                 "Pool mode parameter only support 'max', 'sum',"
                 " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
         self.mode = mode
+        super(PoolGrad, self).__init__(openmp=openmp)
 
     def infer_shape(self, node, in_shapes):
         return [in_shapes[0]]
 
 
 class MaxPoolGrad(PoolGrad):
-    def __init__(self, ds, ignore_border, st=None, padding=(0, 0)):
-        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode='max')
+    def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max', openmp=None):
+        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode, openmp)
```

Comment:
We don't want to make the mode available in MaxPoolGrad, as it most be max. This should work:
```
 def __init__(self, ds, ignore_border, st=None, padding=(0, 0), openmp=None):
    PoolGrad.__init__(self, ds, ignore_border, st, padding, mode='max', openmp=openmp)
```

I removed the double argument mode and I made openmp a named argument.

```
 def __init__(self, ds, ignore_border, st=None, padding=(0, 0), openmp=None):
    PoolGrad.__init__(self, ds, ignore_border, st, padding, 'max', openmp)
```
should also work.